### DISCUSSION TITLE
NPC Compressed flag

### DISCUSSION THREAD

**Discussion by focustense** [Mutagen-Modding/Mutagen#210] [Q&A] [answered]
I've been using Mutagen for a while now and just noticed that when I copy an NPC record into a new mod (using `GetOrAddAsOverride`), it doesn't have the "Compressed" record flag, even though the original record did.

Is this intentional? Will it matter? I'm not completely sure what the impact of this flag is in the game, but I assume there's a reason why every NPC mod seems to use it?

I tried setting the flag manually, via `SkyrimMajorRecordFlags |= SkyrimMajorRecord.SkyrimMajorRecordFlag.Compressed`, but I don't think that's correct, since when I tried to open up that version of the plugin in xEdit, I got a bunch of error messages such as "Error decompressing [NPC:xxxxxxxx]: [EOutOfMemory] Out of memory>". And I'm definitely not actually out of memory, I think it just sees the records as corrupted. And I see code in Mutagen to decompress, but not to compress, so probably what's happening is that I'm just generating uncompressed records labeled as compressed.

Compression not supported right now?

(P.S. I know the flag is not specific to NPCs, but unlike some record types, NPCs _always_ seem to have this flag in traditional plugins.)

**Comment by Noggog** [comment, marked as answer]
Compression going out is not currently supported by Mutagen.  It could be if someone put the work into it.  Most of the work would be making sure the right compression settings were used depending on the game, and if that mattered, etc.

It's omitted because it's not too important to have.  As far as I'm aware, compression is used in the base game files just to cut down on space used for the entirety of Skyrim data when downloading, etc.  But for a patcher output, it's not important to re-compress going out as all that does is save a MB or two of diskspace.  Also, if that output is going to be consumed by another patcher/tool, it has to waste time decompressing what we just wrote out.

Just a very low priority feature to have, but nothing fundamentally blocking it from existing

**Reply by focustense** [reply]
Makes sense, a huge "uncompressed" merge is still only a few MB, so I can see why it wouldn't be a priority. Just wasn't sure if the flag was required for anything in particular, other than a negligible decrease in ESP size.
